chemclipse/plugins/org.eclipse.chemclipse.xxd.model/src/org/eclipse/chemclipse/xxd/model/filter/peaks/PercentageAreaFilter.java

```diff
@@ -0,0 +1,144 @@
+
+	@Override
+	public String getName() {
+
+		return "Percentage Area Filter";
+	}
+
+	@Override
+	public String getDescription() {
+
+		return "Filter peaks by percentage area values";
+	}
+
+	@Override
+	public Class<PercentageAreaFilterSettings> getConfigClass() {
+
+		return PercentageAreaFilterSettings.class;
+	}
+
+	@Override
+	public boolean acceptsIPeaks(Collection<? extends IPeak> items) {
+
+		return true;
+	}
+
+	@Override
+	public <X extends IPeak> void filterIPeaks(CRUDListener<X, IPeakModel> listener, PercentageAreaFilterSettings configuration, MessageConsumer messageConsumer, IProgressMonitor monitor) throws IllegalArgumentException {
+
+		Collection<X> read = listener.read();
+		if(configuration == null) {
+			configuration = createConfiguration(read);
+		}
+		SubMonitor subMonitor = SubMonitor.convert(monitor, read.size());
+		double areaSum = calculateAreaSum(read);
+
+		for(X peak : read) {
+			double compareAreaValue = calculatePercentageAreaCompareValue(peak, areaSum);
+			applySelectedOptions(compareAreaValue, configuration, listener, peak);
+			checkForKeepOption(listener, configuration, peak);
+			subMonitor.worked(1);
+		}
+	}
+
+	private static <X extends IPeak> double calculateAreaSum(Collection<X> read) {
+
+		double areaSum = 0;
+		for(X peak : read) {
+			areaSum = areaSum + peak.getIntegratedArea();
+		}
+		return areaSum;
+	}
+
+	private static <X extends IPeak> double calculatePercentageAreaCompareValue(X peak, double areaSum) {
+
+		return (100 / areaSum) * peak.getIntegratedArea();
+	}
+
+	private static <X extends IPeak> void applySelectedOptions(double peakValue, PercentageAreaFilterSettings localSettings, CRUDListener<X, IPeakModel> listener, X peak) {
+
+		switch (localSettings.getFilterSelectionCriterion()) {
+		case AREA_LESS_THAN_MINIMUM:
+			if(Double.compare(peakValue, localSettings.getMinimumPercentageAreaValue())<0) {
+				processPeak(listener, localSettings, peak);
+			}
+			break;
+		case AREA_GREATER_THAN_MAXIMUM:
+			if(Double.compare(peakValue, localSettings.getMaximumPercentageAreaValue())>0) {
+				processPeak(listener, localSettings, peak);
+			}
+			break;
+		case AREA_NOT_WITHIN_RANGE:
+			if(Double.compare(peakValue, localSettings.getMinimumPercentageAreaValue())<0 || Double.compare(peakValue, localSettings.getMaximumPercentageAreaValue())>0) {
+				processPeak(listener, localSettings, peak);
+			}
+			break;
+		default:
+			throw new IllegalArgumentException("Unsupported Peak Filter Selection Criterion!");
+		}
+	}
+
+	private static <X extends IPeak> void processPeak(CRUDListener<X, IPeakModel> listener, PercentageAreaFilterSettings localSettings, X peak) {
+
+		switch (localSettings.getFilterTreatmentOption()) {
+		case ENABLE_PEAK:
+			peak.setActiveForAnalysis(true);
+			listener.updated(peak);
+			break;
+		case DEACTIVATE_PEAK:
+			peak.setActiveForAnalysis(false);
+			listener.updated(peak);
+			break;
+		case KEEP_PEAK:
```

Comment:
I dont understand this. Why do you modify here the active for analysis property?
As far as I can tell, it should be:
if peak matches filter criterion, do nothing. Else, apply "Delete peak" option.

Reply:
The idea behind this is that the peak to be kept should be differentiated from the peak to be deleted. Therefore, the property is first modified and then the marked peaks are removed. In this way, the peaks sought are retained.

If I think about it again, this approach is not entirely optimal.

Reply:
According to Philip, the requirement is that the peaks that are not kept should also be deleted.